frontend/src/app/pipes/humanize-enum.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'humanizeEnum'
+})
+export class HumanizeEnumPipe implements PipeTransform {
+
+  transform(value: string): any {
+    if ((typeof value) !== 'string') {
+      return value;
+    }
+    value = value.split('_').join(' ');
+    value = value.toLowerCase();
+    return value.charAt(0).toUpperCase() + value.slice(1);
```

Comment:
just to confirm, does this successfully handle the case of empty string?

Reply:
Good point, will add test for that and check.

Reply:
Added the test and now also trims the string.